ast/src/generic.rs

```diff
@@ -317,6 +317,32 @@ pub enum Stmt {
     Break(StmtBreak),
     #[is(name = "continue_stmt")]
     Continue(StmtContinue),
+
+    // Jupyter notebook specific
+    #[is(name = "line_magic_stmt")]
+    LineMagic(StmtLineMagic),
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct StmtLineMagic {
+    pub range: TextRange,
+    pub kind: String,
+    pub value: String,
+}
```

Comment:
I'm thinking of using a enum for the `kind` field similar to the `Constant` enum.

Reply:
Using an enum might not be possible as there are no individual tokens for the magic kinds.

Reply:
can you explain what you mean by "no individual tokens". Can't we use an enum similar to `MagicKind` in the lexer?

Can you document the kind field if we cannot use an enum so that users know what it contains?

Reply:
Moved the `MagicKind` enum to the AST module and reusing it in the node.